### DISCUSSION TITLE
EGSnrc, Problems getting started with myapp

### DISCUSSION THREAD

**Discussion by UweHaeusler** [nrc-cnrc/EGSnrc#1056] [Questions]
Dear community.

I have installed EGSnrc with mingw64 on a 64bit Win10-system and the installation didn't show no error...
After trying my first simple "myapp" following exactly the tutorial "getting-started.pdf", I get the following result after typing "mingw32-make.exe":

```
_C:\EGSnrc\EGS_HOME\myapp>mingw32-make.exe
/usr/bin/bash: -c: line 2: syntax error: unexpected end of file
g++ -I. -IC:\EGSnrc\HEN_HOUSE\lib\EGSNRC_Win64 -IC:\EGSnrc\HEN_HOUSE\egs++ -IC:\EGSnrc\HEN_HOUSE\interface -DWIN32 -fPIC  -O2 -mtune=native -c -o myapp_EGSNRC_Win64.o myapp.cpp
g++ -I. -IC:\EGSnrc\HEN_HOUSE\lib\EGSNRC_Win64 -IC:\EGSnrc\HEN_HOUSE\egs++ -IC:\EGSnrc\HEN_HOUSE\interface -DWIN32 -fPIC  -O2 -mtune=native -c -o egs_advanced_application_EGSNRC_Win64.o C:\EGSnrc\HEN_HOUSE\egs++\egs_advanced_application.cpp
Mortran compiling EGSnrc sources ...
gfortran -cpp -O2 -mtune=native -fPIC    -DCOMPILE_TIME="\"21.08.2023  11:36\"" -DGIT_HASH="\"unknown\"" -c -o egsnrc_EGSNRC_Win64.o egsnrc_EGSNRC_Win64.F
g++ -I. -IC:\EGSnrc\HEN_HOUSE\lib\EGSNRC_Win64 -IC:\EGSnrc\HEN_HOUSE\egs++ -IC:\EGSnrc\HEN_HOUSE\interface -DWIN32 -fPIC  -O2 -mtune=native -c -o egs_interface2_EGSNRC_Win64.o C:\EGSnrc\HEN_HOUSE\interface\egs_interface2.c
g++ -O2 -mtune=native -o C:\EGSnrc\EGS_HOMEbin\EGSNRC_Win64\myapp.exe myapp_EGSNRC_Win64.o egs_advanced_application_EGSNRC_Win64.o egsnrc_EGSNRC_Win64.o egs_interface2_EGSNRC_Win64.o -LC:\EGSnrc\HEN_HOUSE\egs++\dso\EGSNRC_Win64 -legspp -lgfortran
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/7.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot open output file C:\EGSnrc\EGS_HOMEbin\EGSNRC_Win64\myapp.exe: No such file or directory
collect2.exe: error: ld returned 1 exit status
mingw32-make.exe: *** [C:\EGSnrc\HEN_HOUSE\makefiles\cpp_makefile:44: C:\EGSnrc\EGS_HOMEbin\EGSNRC_Win64\myapp.exe] Error 1_
```

Obviously, the g++ compiler cannot generate the file myapp.exe because it is using the wrong directory "C:\EGSnrc\EGS_HOMEbin\EGSNRC_Win64\" instead of "C:\EGSnrc\EGS_HOME\bin\EGSNRC_Win64\" (missing the backslash between EGS_Home and bin). I already checked the environment variables - they are set correctly.

Any ideas for solving the problem are appreciated.

Thanks in advance
Uwe

**Comment by rtownson** [comment]
Does your EGS_HOME environment variable end with a slash? 

When did you download EGSnrc? There was an update for v2023a released on August 3rd, so I'd like to know if it was before or after that.

I noticed that you didn't install using git. Please consider installing git and using 'git clone' instead of downloading the zip file to install EGSnrc. This makes it much easier to track what is going on. You may not need to re-install yet, but it is likely a good solution to delete everything and start over using git.

**Reply by UweHaeusler** [reply]
Thanks for your answer. Unfortunately, I haven't checked my inbox recently and have not noticed your answer already. I did this installation with the latest 2023a-release of EGSnrc (as mentioned by you), but I didn't have no success at all ...

Anyway, I was pretty unhappy with the problem! Thus, I started a brandnew installation recently following this recommendation https://www.youtube.com/watch?v=ZFBYh7diO8w, because I have a Windows 10 Enterprise LTSC. Therefore, I have installed mingw64 vers. 6.3.0 and the 2020 version (incl. the configure-windows.exe) of EGSnrc (https://github.com/nrc-cnrc/EGSnrc/releases/tag/v2020). After defining the environment variables I had succes with this combination, namely 
- I build a myapp.exe and used it with different slabs and 
- I successfully compiled tutor7pp and used it with different test data.

But then ... I wanted to build dosrznrc.exe (this application is most interesting for my physical question) and it failed. The system says:

```
_C:\EGSn20\EGS_HOME\dosrznrc>mingw32-make
Mortran compilation for dosrznrc_win3264.F
mortran3.exe: using following I/O units
  raw/hex data file: C:\EGSn20\HEN_HOUSE\bin\win3264\mortran3.dat
  command line files:
  -> appending C:\EGSn20\HEN_HOUSE\src\egsnrc.macros
  -> appending C:\EGSn20\HEN_HOUSE\utils\timing.macros
  -> appending C:\EGSn20\HEN_HOUSE\lib\win3264\machine.macros
  -> appending C:\EGSn20\HEN_HOUSE\src\ranmar.macros
  -> appending C:\EGSn20\HEN_HOUSE\src\transportp.macros
  -> appending C:\EGSn20\HEN_HOUSE\src\pegs4_macros.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\phsp_macros.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\iaea_phsp_macros.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\srcrz.macros
  -> appending dosrznrc.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\ranmar.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\srcrz.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\ensrc.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\geomrz.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\get_inputs.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\get_media_inputs.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\grids.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\nrcaux.mortran
  -> appending C:\EGSn20\HEN_HOUSE\utils\xvgrplot.mortran
  -> appending C:\EGSn20\HEN_HOUSE\lib\win3264\machine.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\egs_utilities.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\egs_parallel.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\pegs4_routines.mortran
  -> appending C:\EGSn20\HEN_HOUSE\src\egsnrc.mortran
  generated mortjob.mortran
  fortran output: dosrznrc_win3264.F
  mortlst output: dosrznrc_win3264.mortlst

  Mortran compiling ...  OK
process_begin: CreateProcess(NULL, echo ., ...) failed.
make (e=2): Das System kann die angegebene Datei nicht finden.
mingw32-make: *** [C:\EGSn20\HEN_HOUSE\makefiles\standard_makefile:87: C:\EGSn20\EGS_HOME\bin\win3264\dosrznrc.exe] Error 2_
```

I suppose, that this is a FORTRAN code (different from the other programs, which were C++) and the system does not use the correct compiler command? 

I have attached my log-file and a scrennshot of my environment varaiables. I'd appreciate very much your help!

[config_win3264.log](https://github.com/nrc-cnrc/EGSnrc/files/12456369/config_win3264.log)
![Env-var](https://github.com/nrc-cnrc/EGSnrc/assets/142793216/51538e34-72ee-4457-9fff-74740d74298b)

**Comment by rtownson** [comment]
If you look in the installation log there are many failures, both for the C++ and Fortran codes. I noticed that you installed mingw 6.4.0, not 6.3.0. I think that 6.4.0 should work, but I wanted to check to make sure that you haven't installed multiple versions, since that could cause problems. Make sure that your old installation of mingw in `C:/mingw64` is not causing problems. Otherwise, I'm not sure what the issues you are seeing are from. Now that you're using a different version of mingw and git, you could give v2023a another try (just make sure to update your environment variables).

**Reply by UweHaeusler** [reply]
Thanks for your help. You were right: it was mingw 6.4.0 installed. I'm sorry for asking again, but I still have (even more!) problems ...

I gave the whole system a new try (as recommended):
- I deleted the old mingw in C:/mingw64 and my old EGSnrc in C:\EGSn20
- I tried the command "mingw32-make --version" and it gave correct results
- i deleted the old environment variables
- I downloaded EGSnrc newly using git (unfortunately, it doesn't say which EGSnrc it is downmoading?): it didn't show no error and I renamed the directory to C:\EGSn23\
- I downloaded https://github.com/nrc-cnrc/EGSnrc/releases/download/v2023/EGSnrc-configure-windows.exe (there is no extra version for  2023a?) and executed it (incl. env. variables and gui links), it appered to work correctly (log-file is attached)
- I restarted Win10
[config_win3264.log](https://github.com/nrc-cnrc/EGSnrc/files/12547218/config_win3264.log)

Then, I checked the system (all env. variables are correct, make-command worked ...) and started again with the first simple program "myapp" following the getting-started.pdf. And ... it failed saying:
"C:\EGSn23\EGS_HOME\myapp>mingw32-make
Makefile:69: makefilescpp_makefile: No such file or directory
mingw32-make: *** No rule to make target 'makefilescpp_makefile'.  Stop."
Obviously, the line 69 (include $(HEN_HOUSE)makefiles$(DSEP)cpp_makefile) didn't work, esp. there are problems with the $(DSEP)-function (replacing "/" or "\")? I really don't understand, why this problem now occurs, whilst it it didn't occur in my last installation (s. above)? Therefore, I've compared the files win3264.conf of both installations and there is no difference (apart from the date and the main directory)? 

I am really desperate! I have tried so many versions and since now the previous combination of mingw64 vers. 6.4.0 and the 2020 version of EGSnrc (using https://github.com/nrc-cnrc/EGSnrc/releases/download/v2020/EGSnrc-configure-windows.exe) was the only installation, that worked at least partly (s. above). But I remember, that I used EGSnrc 15-20 years ago on a Win XP system and when I downloaded and installed it then it was pretty straight forward and worked fully from the beginning ...

Anyways, any help in order to get a working installation of EGSnrc on my Win10 system is appreciated. Many thanks in advance.

**Reply by UweHaeusler** [reply]
I have another idea, because I also could use a virtual machine: 
Is there a fully working EGSnrc version (i.e. on a linux system) completely build as iso-file?
Thanks for all your help.

**Reply by rtownson** [reply]
Currently I'm not sure the issue behind your installation problem, but we'll look into it. For now, yes using a linux virtual machine is a great idea. It should install very smoothly using the instructions here:
https://github.com/nrc-cnrc/EGSnrc/wiki/Install-EGSnrc-on-VirtualBox

**Reply by UweHaeusler** [reply]
Thanks for your answer and sorry for responding late. 
Meanwhile I have moved the installation to another Windows 10 PC, where I have installed the EGSnrc package 2023a according to your instructions and it's fully working (I suppose, the OS of the former laptop was partly corrupted?)!

But, after doing my first calculations, now I have some physical questions:
I compiled and used dosrznrc, because I am interested in calculations of NORM material from a disc source. In order to understand the EGS system and to compare with known values I was trying some simple models of a Co-60 and a Cs-137 point source (1 mm x 1mm) made from iron and calculated the dose in a 10 cm x 10 cm cylinder of ICRU tissue (in front of more ICRU tissue) at a distance of 95-105 cm (average: 100 cm) from the source. My results with EGSnrc were: 
Co-60  3.6E-17 Gy/s/m²/Bq (using HEN_HOUSE\spectra\egsnrc\co60.spectrum)
Co-60  4.3E-17 Gy/s/m²/Bq (using HEN_HOUSE\spectra\egsnrc\bareco60.spectrum)
Cs-137 2.2E-17 Gy/s/m²/Bq (using HEN_HOUSE\spectra\egsnrc\cs137.spectrum)
Corresponding dose rates in literature (i.e. Ioan et al, "Co-60 SPECIFIC GAMMA-RAY CONSTANT DETERMINATIONS FOR VARIOUS BIOLOGICAL MATERIALS INVOLVED IN RADIOTHERAPY PROCEDURES BY USING GEANT4 AND NIST XCOM", Romanian Journal of Physics 63, 701, 2018) are:
Co-60  9.4E-17 Gy/s/m²/Bq
Cs-137 2.5E-17 Gy/s/m²/Bq

At first, I don't understand the data in the file co60.spectrum, which differ significantly from a Co-60 spectrum according to literature. The data in the file bareco60.spectrum are simplified but they correspond better to the Co-60 decay scheme (and gave better but not sufficient results ...). Furthermore, my results differ very much (esp. for Co-60) from data from literature? 
Thus, I am not sure, if I did understand the input format correctly (radius r and height z according to a cylindrical coordinate system)?

I have attached my corresponding egslst-files and I would appreciate very much, if you could give me some information, what is wrong with my calculations. (maybe I should primarily take part at a training course, but I wanted to gain results soon, I'm sorry ...). 

With many thanks in advance.

[Cs-137-Source-small_1m_inAIR_tissue.egslst.txt](https://github.com/nrc-cnrc/EGSnrc/files/12675381/Cs-137-Source-small_1m_inAIR_tissue.egslst.txt)
[BareCo60-Source-small_1m_inAIR_tissue.egslst.txt](https://github.com/nrc-cnrc/EGSnrc/files/12675382/BareCo60-Source-small_1m_inAIR_tissue.egslst.txt)
[Co60-Source-small_1m_inAIR_tissue.egslst.txt](https://github.com/nrc-cnrc/EGSnrc/files/12675383/Co60-Source-small_1m_inAIR_tissue.egslst.txt)

**Reply by rtownson** [reply]
From a quick glance at the paper, it seems like they don't tell you the actual spectrum of Co-60 that they used, and whether it included the shielding effects of the Co-60 capsule that they used in the experiment. It seems challenging to get good agreement without that information. Also, did you confirm that the material compositions used in EGSnrc are in perfect agreement with those used in Geant4? If not, you should create new materials to match.
